extensions/worker-specific-hashrate-tracking.md

```diff
@@ -0,0 +1,105 @@
+
+Terms like "MUST," "MUST NOT," "REQUIRED," etc., follow RFC2119 standards.
+
+---
+
+## 1. Extension Overview
+
+This extension enables mining pools to track individual workers (`user_identity`) within shared channels by introducing a `user_id` field within share submission messages. This implementation can be achieved through one of two options: (1) extending the existing `SubmitSharesExtended` message to include `user_id`, or (2) creating a new message, `SubmitIdentifiedSharesExtended`, with `user_id` integrated for explicit worker tracking. Additionally, the extension includes `SetUserIdentity` messages that allow clients and servers to map each `user_id` to a unique `user_identity`, ensuring accurate tracking of worker hashrates across extended channels.
+
+### 1.1 Activate Extension
+
+A client wishing to use this extension sends the `Activate` message. If supported, the pool responds with `Activate.Success`. Otherwise, the client stops further attempts.
+
+---
+
+## 2. Extension Messages
+
+### `Activate` (Client -> Server)
+
+| Field Name | Data Type | Description                                                 |
+| ---------- | --------- | ----------------------------------------------------------- |
+| request_id | U32       | Unique identifier for pairing the response                  |
+
+### `Activate.Success` (Server -> Client)
+
+| Field Name | Data Type | Description                                                 |
+| ---------- | --------- | ----------------------------------------------------------- |
+| request_id | U32       | Unique identifier for pairing the response                  |
+
+### `SetUserIdentity` (Client -> Server)
+
+The client synchronizes the `user_id` ↔ `user_identity` mapping with the server (Pool).
+
+| Field Name    | Data Type        | Description                           |
+|---------------|------------------|---------------------------------------|
+| channel_id    | U32              | Channel identifier                    |
+| request_id    | U32              | Unique request ID                     |
+| user_id       | SEQ0_64K[U32]    | Unique IDs for each worker            |
+| user_identity | SEQ0_64K[STR0_255]| Worker identities (names)            |
+
+### `SetUserIdentity.Success` (Server -> Client)
+
+Response indicating the pool has accepted the mapping.
+
+| Field Name    | Data Type    | Description                               |
+|---------------|--------------|-------------------------------------------|
+| channel_id    | U32          | Channel ID                                |
+| request_id    | U32          | Original request ID                       |
+| user_id       | SEQ0_64K[U32]| Worker ID associated                      |
+
+### `SetUserIdentity.Error` (Server -> Client)
```

Comment:
Here the pool should close the connection because it means something is not working properly or client is trying to spam the server. Does it make sense?